fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ResolveOrdinalInOrderByAndGroupBy.java

```diff
@@ -0,0 +1,96 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.analysis;
+
+import org.apache.doris.nereids.properties.OrderKey;
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.rules.expression.rewrite.rules.FoldConstantRule;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.literal.IntegerLikeLiteral;
+import org.apache.doris.nereids.trees.expressions.literal.Literal;
+import org.apache.doris.nereids.trees.plans.logical.LogicalAggregate;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * SELECT col1, col2 FROM t1 ORDER BY 1 -> SELECT col1, col2 FROM t1 ORDER BY col1
+ * SELECT col1, SUM(col2) FROM t1 GROUP BY 1 -> SELECT col1, SUM(col2) FROM t1 GROUP BY col1
+ */
+public class ResolveOrdinalInOrderByAndGroupBy implements AnalysisRuleFactory {
+
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.<Rule>builder()
+                .add(RuleType.ResolveOrdinalInOrderBy.build(
+                        logicalSort().then(sortNode -> {
+                            List<Slot> childOutput = sortNode.child().getOutput();
+                            List<OrderKey> orderKeys = sortNode.getOrderKeys();
+                            List<OrderKey> orderKeysWithoutConst = new ArrayList<>();
+                            for (OrderKey k : orderKeys) {
+                                Expression expression = k.getExpr();
+                                expression = FoldConstantRule.INSTANCE.rewrite(expression);
+                                if (expression instanceof IntegerLikeLiteral) {
+                                    IntegerLikeLiteral i = (IntegerLikeLiteral) expression;
+                                    int ord = i.getIntValue();
+                                    checkOrd(ord, childOutput.size());
+                                    orderKeysWithoutConst
+                                            .add(new OrderKey(childOutput.get(ord - 1), k.isAsc(), k.isNullFirst()));
+                                } else {
+                                    orderKeysWithoutConst.add(k);
+                                }
+                            }
+                            return sortNode.withOrderByKey(orderKeysWithoutConst);
```

Comment:
if no ordinal in orderkeys, we should return original object for fast equals to reduce plan time